File: models/Usuario.cs
namespace TallerMecanico.models;

public class Usuario
{
    public int Id { get; set; }
    public string Nombre { get; set; }
    public string Apellido { get; set; }
    public string Email { get; set; }
    public string PasswordHash { get; set; }
    public RolUsuario Rol { get; set; } // Admin, Cliente
    public DateTime FechaRegistro { get; set; }
    public bool Estado { get; set; }

    // Auditoría
    public DateTime FechaCreacion { get; set; }
    public DateTime? FechaActualizacion { get; set; }
    public DateTime? FechaBorrado { get; set; } 
    public bool EstaBorrado { get; set; } = false;
}

public enum RolUsuario
{
    Admin,
    Cliente
}

File: AFront/src/app/Entity/Usuario.ts
export interface Usuario {
  id: number;
  nombre: string;
  apellido: string;
  email: string;
  rol: RolUsuario;
  fechaRegistro: Date;
  estado: boolean;
  fechaCreacion: Date;
  fechaActualizacion: Date | null;
  fechaBorrado: Date | null;
  estaBorrado: boolean;
}

export enum RolUsuario {
  Admin = 0,
  Cliente = 1
}
